File: java/datastructure/datastructure/linkedList/LinkedListNode.java
package datastructure.linkedList;

/***
 * @class : LinkedListNode.java 
 * SinglyLinkedListNode
*/



public class LinkedListNode<T> {
    private T data; 
    private LinkedListNode<T> next; 
    
    public LinkedListNode() {
        this.next = null; 
    }
    
    public LinkedListNode(T value) {
        this.data = value;
        this.next = null;  
    }
   
    public T getData() {
        return this.data; 
    }

    public LinkedListNode<T> getNext() {
        return this.next; 
    }

    public void setData(T value) {
        this.data = value; 
    }

    public void setNext(LinkedListNode<T> link) {
        this.next = link; 
    }

}

File: cpp/datastructure/datastructure/LinkedList/LinkedListNode.cpp
#ifndef LINKEDLISTNODE_H 
#define LINKEDLISTNODE_H 

template <typename T> 
class SinglyLinkedListNode {
    public: 
        SinglyLinkedListNode(); 
        SinglyLinkedListNode(T element); 
        void setData(T element); 
        T getData(); 
        SinglyLInkedListNode<T> getNext() {
            return this->next; 
        }
        void setNext(SinglyLinkedListNode<T> next_node); 

    private: 
        SinglyLinkedListNode<T> next; 
        T data; 

}; 


template <typename T> 
SinglyLinkedListNode<T>::SinglyLinkedListNode() {
    this.next = nullptr; 
}

template <typename T> 
SinglyLinkedListNode<T>::SinglyLinkedListNode(T element) {
    this->data = element; 
    this.next = nullptr; 
}

template <typename T> 
void 




#endif 